### PR TITLE
DATA-3097: Update training script for detection to take in command line args

### PR DESCRIPTION
**Pull request by vpandiarajan20**
Adds labels and epochs as arguments to training script so the user can choose the values
Changed num epochs to 3, changed labels to green square and blue star
Screenshot 2024-09-26 at 3 55 52 PM
Screenshot 2024-09-26 at 3 56 19 PM
Screenshot 2024-09-26 at 3 57 18 PM

> `vignesh.pandiarajan@viam.com@ROBOT-GJ3J9QVM9L rdk % go run cli/viam/main.go train submit custom with-upload --dataset-id=6515b62ba420eb685e988e83 --model-org-id=0fbe951e-d4c6-427f-985f-784b7b85842c --model-name=vig-detection-C --model-type=object_detection --framework=tflite --path=/Users/vignesh.pandiarajan@viam.com/viam-modules-originals/cli-args-detection.tar.gz --script-name=versionC --args=num_epochs=3,labels="'green_square blue_star'"`

### CODE REVIEW

**Comment by vpandiarajan20** [comment]
Changed num epochs to 175, changed labels to green square and blue star
image

image
Screenshot 2024-09-23 at 2 31 48 PM

File: model/training.py [hunk 1]
```diff
@@ -28,8 +28,15 @@ def parse_args():
     parser.add_argument("--dataset_file", dest="data_json", type=str)
     parser.add_argument("--model_output_directory", dest="model_dir", type=str)
     parser.add_argument("--num_epochs", dest="num_epochs", type=int)
+    parser.add_argument(
+        "--labels",
+        dest="labels",
+        type=str,
+        required=False,
+        help="Slash(/)-separated list of labels",
```
Comment (tahiyasalam) [inline model/training.py:36]: Instead of a slash separate list, we can just nargs in argparse. Here's an example https://stackoverflow.com/a/32763023

File: scripts/test.sh [hunk 2]
```diff
@@ -4,4 +4,11 @@ python3 setup.py sdist --formats=gztar
 pip install dist/model-0.1.tar.gz
 
 python3 -m model.training --dataset_file=dataset.jsonl --model_output_directory=. --num_epochs=2
-pip install pytest && pytest tests/
+pip install pytest && pytest tests/test_file_exists.py
+
+rm labels.txt
+rm detection.tflite
+
+python3 -m model.training --dataset_file=dataset.jsonl --model_output_directory=. --num_epochs=2 --labels=green_square/blue_star
```
Comment (tahiyasalam) [inline scripts/test.sh:12]: I don't think we should actually need to execute the code in order to test the args parsing. Instead, we should probably refactor parse_args to take in args, instead of relying on the implicit use the default [sys.argv](https://docs.python.org/3/library/sys.html#sys.argv). The docs describe this in a little bit more detail [here](https://docs.python.org/3/library/argparse.html#parsing-arguments). Then, in the main, we can call `parse_args(sys.argv[1:])`. In the tests, we would just explicitly check that the labels are being parsed as expected & nothing else.

File: tests/test_label_change.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import os
+
+
+def test_label_change():
```
Comment (tahiyasalam) [inline tests/test_label_change.py:4]: Related to the comment above, I think we should just be testing the args parser.

**Review by tahiyasalam — CHANGES_REQUESTED** [review]
Thanks so much for taking this on! Some comments on testing + using some nice formatting functionality built in to the args parser.

File: model/training.py [hunk 4]
```diff
@@ -20,16 +21,28 @@
...
     parser.add_argument("--model_output_directory", dest="model_dir", type=str)
     parser.add_argument("--num_epochs", dest="num_epochs", type=int)
-    args = parser.parse_args()
-    return args.data_json, args.model_dir, args.num_epochs
+    parser.add_argument(
+        "--labels",
+        dest="labels",
+        type=str,
+        required=False,
+        help="Space-separated list of labels",
```
Comment (tahiyasalam) [inline model/training.py:37]: Maybe worth saying here that they must be encapsulated by single quotes and leave a comment with an example? Also worth updating the README with instructions on how to submit the job

Comment (tahiyasalam) [reply model/training.py:37]: bumping this! for both this & the classification example

**Review by tahiyasalam — APPROVED** [review]
Great work!! Thank you so much
